vllm/engine/llm_engine.py

```diff
@@ -956,7 +956,8 @@ def _check_stop(self, seq: Sequence,
                 return
         if seq.get_last_token_id() in sampling_params.stop_token_ids:
             stop_str = self.get_tokenizer_for_seq(seq).convert_ids_to_tokens(
-                seq.get_last_token_id())
+                [seq.get_last_token_id()],
```

Comment:
Curious - why do we need to make this a list?

Reply:
Because I found in the tokenizer, if we only pass an int, `convert_ids_to_tokens` would not use `skip_special_tokens`:

https://github.com/huggingface/transformers/blob/638c423c89a7996dd5508f228ac2943a743673de/src/transformers/tokenization_utils_fast.py#L386

Reply:
Oh wow, good find! (This does also seem a bug to me on `transformers`?)

Reply:
Logically it seems a bug, but I guess they may not expect users to pass a `special_token_id` int with `skip_special_tokens` set to `true`